### ISSUE TITLE
Bluebird promises bind not working properly

### ISSUE THREAD

**Issue by LeonardoGentile** [balderdashy/waterline#1232]
If I use
```
User.findOne({email:data.email})
  .bind({})
  .then(function(user) {
   //...
```
Things won't work `TypeError: Object [object Object] has no method 'bind'`
But If i put the `bind` after the first `then` then things will work as expected

```
User.findOne({email:data.email})
  .then(function(user) {})
  .bind({}) 
  //...
```

Maybe the promise chain is wrapped as bluebird promise only after the first `then`?

**Comment by devinivy** [comment]
This is true.  Currently only the most basic promise methods are supported.  This was originally for ease marrying the existing `User.find().exec(cb)` ( also, `.populate()`, `.where()`, etc.) syntax with promise support, but I think we can deal with this more gracefully to support the full bluebird API.  Would you like to make a PR to our roadmap file to put this in the queue?

**Comment by LeonardoGentile** [comment]
@devinivy :+1: done

**Comment by devinivy** [comment]
Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @LeonardoGentile.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by kirtangajjar** [comment]
Is it solved? Really?
